docs/pull_request_template.md

```diff
@@ -0,0 +1,38 @@
+### All Submissions:
+
+* [ ] Have you followed the guidelines in our **CONTRIBUTING** document?
+* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
+
+<!-- You can erase any parts of this template not applicable to your Pull Request. -->
+
+
+<!-- These have been added according to the rOpenSci guidelines. -->
+<!-- From: https://www.talater.com/open-source-templates/#/page/99 -->
+
+## Types of changes
+<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
+- [ ] Bug fix (non-breaking change which fixes an issue. List the bug ID if applicable)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing core functionality to change) 
+
+
+### New Feature Submissions checklist:
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
+1. [ ] Have you ensured there are no conflicts with major packages (e.g. tidyverse, MASS) or base R packages?
+2. [ ] Does your new feature require documentation, and if so, have you udpated the documentation?
+3. [ ] Have you added tests for your features, and do your features pass all tests?
+4. [ ] Have you linted your code with lintr locally prior to submission?
+5. [ ] Have you styled your code with styler locally prior to submission?
+6. [ ] Have you checked the performance of your code?
+7. [ ] Does your code comply with rOpenSci guidelines?
+8. [ ] Does your code comply with CRAN policies?
+
```

Comment:
Can we add links to both of these?

Reply:
7. This one: https://devguide.ropensci.org/? or specifcially this one: https://devguide.ropensci.org/guide-for-authors.html?
8. this one: https://cran.r-project.org/web/packages/policies.html? 

These are not really checklists, should we create a further checklist?

Reply:
It's probably asking a lot from the users to have them read extensive external guidelines. I suspect they'd reply yes even if they'd not checked over the policies... Specific issues will have to be identified in review.

Reply:
@jnothman that's a fair point. Should we exclude points 7 and 8 then? Do we usually assume contributors are at least aware of these policies, or is it up to the package owners themselves to make sure incoming code is compliant?

Reply:
> Should we exclude points 7 and 8 then? Do we usually assume contributors are at least aware of these policies, or is it up to the package owners themselves to make sure incoming code is compliant?

I think if your contrib guidelines say that you'd like new code to be compliant with these best practices, and refer the contributor to them, it's probably sufficient to then be supplemented by explicit reveiw.

Reply:
Sounds good. Perhaps if there's anything particularly prominent in those guidelines not covered in this PR template, then we can add them to the template?

Reply:
I think the key parts of the guidelines (e.g. styling, cross-platform compatibility, roxygen2 - the latter of which will be implemented) are covered in this PR template. There's nothing else major popping out to me in the CRAN/rOpenSci guidelines that we need to include here. Since `CONTRIBUTING.md` already links potential contributors to these guidelines, so I think it's safe to remove points 7 and 8.